source/agora/node/Node.d

```diff
@@ -230,6 +234,28 @@ public class Node : API
         return new DiskMetadata(data_dir);
     }
 
+    /***************************************************************************
+
+        Returns an instance of a BlockStorage or MemoryStorage
+
+        Note: not exposed in the API.
+
+        Params:
+            data_dir = path to the blockdata directory
+
+        Returns:
+            Returns instance of `MemoryStorage` if data_dir is empty,
+            otherwise returns instance of `BlockStorage`
+
+    ***************************************************************************/
+
+    protected IBlockStorage getBlockStorage (string data_dir) @system
+    {
+        if (data_dir == "")
+            return new MemBlockStorage();
```

Comment:
It shouldn't be based on the runtime data, because someone could make a mistake and pass a string which is `null` by accident. We usually use `version (unittest)` or override.